File: data/obligaciones.py
from dataclasses import dataclass
from typing import List


@dataclass
class Obligacion:
    id: str
    nombre: str
    periodicidad: str
    descripcion: str


OBLIGACIONES: List[Obligacion] = [
    Obligacion(
        id="renta_pj",
        nombre="Declaración de Renta",
        periodicidad="Anual",
        descripcion="Personas jurídicas — declaración y pago en dos cuotas (mayo y julio).",
    ),
    Obligacion(
        id="iva_bimestral",
        nombre="IVA Bimestral",
        periodicidad="Bimestral",
        descripcion="Declaración y pago del Impuesto al Valor Agregado cada dos meses.",
    ),
    Obligacion(
        id="retencion",
        nombre="Retención en la Fuente",
        periodicidad="Mensual",
        descripcion="Presentación y pago mensual de retenciones practicadas a terceros.",
    ),
    Obligacion(
        id="pila",
        nombre="Seguridad Social (PILA)",
        periodicidad="Mensual",
        descripcion="Aportes a salud, pensión, ARL y parafiscales vía Planilla Integrada de Liquidación de Aportes.",
    ),
    Obligacion(
        id="ica",
        nombre="ICA / ReteICA",
        periodicidad="Bimestral / Anual",
        descripcion="Impuesto de Industria y Comercio — periodicidad y tarifa según municipio.",
    ),
    Obligacion(
        id="exogena",
        nombre="Información Exógena",
        periodicidad="Anual",
        descripcion="Reporte de información tributaria a la DIAN (terceros, socios, pagos, etc.).",
    ),
]

OBLIGACIONES_MAP: dict = {o.id: o for o in OBLIGACIONES}

File: src/data/obligaciones.ts
export interface Obligacion {
  id: string;
  nombre: string;
  nombreEn: string;
  periodicidad: 'mensual' | 'bimestral' | 'anual';
  categoria: 'tributaria' | 'laboral';
}

export const obligaciones: Obligacion[] = [
  {
    id: 'renta_pj',
    nombre: 'Declaración de Renta — Persona Jurídica',
    nombreEn: 'Corporate Income Tax Return',
    periodicidad: 'anual',
    categoria: 'tributaria',
  },
  {
    id: 'iva_bimestral',
    nombre: 'IVA Bimestral',
    nombreEn: 'Bimonthly VAT',
    periodicidad: 'bimestral',
    categoria: 'tributaria',
  },
  {
    id: 'retencion',
    nombre: 'Retención en la Fuente',
    nombreEn: 'Withholding Tax',
    periodicidad: 'mensual',
    categoria: 'tributaria',
  },
  {
    id: 'pila',
    nombre: 'Seguridad Social (PILA)',
    nombreEn: 'Social Security (PILA)',
    periodicidad: 'mensual',
    categoria: 'laboral',
  },
];

export const obligacionesMap: Record<string, Obligacion> = Object.fromEntries(
  obligaciones.map((o) => [o.id, o])
);
